apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl

```diff
@@ -378,6 +395,37 @@ is_auth_key(Key) when is_binary(Key) ->
 is_auth_key(_) ->
     false.
 
+get_auth_uri(#{version := v1}, Config) ->
+    Username = proplists:get_value(username, Config, "undefined"),
+    Password = proplists:get_value(password, Config, "undefined"),
+    "/query?" ++
+        uri_string:compose_query([
+            {"u", Username},
+            {"p", Password},
+            {"q", "show databases"}
+        ]);
+get_auth_uri(#{version := v2}, _Config) ->
+    "/api/v2/query?" ++
+        uri_string:compose_query([
+            {"q", "show databases"}
+        ]).
+
+do_check_auth(Client0, AuthUri) ->
+    Client = Client0#{path := AuthUri},
+    Point = [#{fields => #{<<"check_auth">> => <<"">>}, measurement => <<"check_auth">>}],
+    Ret =
+        case influxdb:write(Client, Point) of
+            {_, {401, _, _}} ->
```

Comment:
I think this authn health check would be better placed at the InfluxDB client itself, since it deals with the internal structure of the client and such.  :thinking:

Reply:
Ack